src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -1500,9 +1500,11 @@ JRT_BLOCK_ENTRY(address, SharedRuntime::handle_wrong_method_ic_miss(JavaThread*
     // Return Method* through TLS
     current->set_vm_result_2(callee_method());
   JRT_BLOCK_END
-  // return compiled code entry point after potential safepoints
-  assert(callee_method->verified_code_entry() != nullptr, " Jump to zero!");
-  return callee_method->verified_code_entry();
+  // Return compiled code entry point after potential safepoints.
+  // In case we are in interpreted only mode return c2i entry instead.
+  address target = current->is_interp_only_mode() ? callee_method->get_c2i_entry() : callee_method->verified_code_entry();
+  assert(target != nullptr, "Jump to zero!");
+  return target;
```

Comment:
There are now 5 copies of this exact code block including this one.

Seems like this is screaming to be refactored into some static function
with an appropriate name that can be called from all five places.

Reply:
I considered that initially but I preferred the readability part given that we are not saving much. But I can do that if you think it's better that way.

Reply:
I'm not totally sold on the refactoring being a better idea. I would be okay if you
decided not to do that. Let's see if any other reviewer has an opinion one way or
the other.

Reply:
I wonder if something like

`callee_method->from_compiled_entry(current->is_interp_only_mode())`

might be readable and well factored.

Reply:
Yes, that works. I can do that but I'll wait since we might not go with this solution after all.

Reply:
> Yes, that works. I can do that but I'll wait since we might not go with this solution after all.

Is there an alternative solution you have in mind?

Reply:
So we can continue with this solution and just special case _linkToNative to always return the verified_code_entry(). The other option would be to just remove the assert in fixup_callers_callsite() added in 8288949 to fix this crash. It doesn't solve the underlying issue where after a change to interpreter only mode we still continue in compiled code, but this is not specific to Continuation.enterSpecial and Continuation.enter as the assert reads. With any caller-callee we face the same issue. This is a long standing bug that is described in 8218403 as you have filed already. 
Now if we continue with this solution the question would be, is _linkToNative the only method that doesn't have an interpreted version? Or are there more special cases to cover?